README.rst

```diff
@@ -537,16 +537,47 @@ the run step:
           namespaced.file.alias2: "/path/to/readwrite/file/or/dir:rw"
           build/dir/file: "/path/to/build/dir/file"
 
-        # A map specifying cache directories that should be mounted inside the
-        # container. The cache directories are maintained between builds and can
-        # be used to store files, such as downloaded dependencies, to speed up
-        # builds. Caches are shared within a build configuration, meaning that
-        # caches with the same name are shared between steps. Caches should be
-        # treated as ephemeral and should only store items that can be
-        # obtained/generated by subsequent builds.
+        # A map specifying cache directories that stored as archive files on the
+        # host system as named <local cache key> and extracted as directory in
+        # the container name <docker path>. The cache directories are maintained
+        # between builds and can be used to store files, such as downloaded
+        # dependencies, to speed up builds. Caches are shared within a build
+        # configuration, meaning that caches with the same name are shared between
+        # steps. Caches should be treated as ephemeral and should only store items
+        # that can be obtained/generated by subsequent builds.
+        #
+        # Two formats are supported when defining caches.
+        # 1) <local cache key>: <docker path> (backwards compatible)
+        #
+        # 2) <docker path>:
+        #      - <local cache key A>
+        #      - <local cache key B>
+        #
+        #    Restore Cache:
+        #      This format allows for prefix matching. The order of the list dictates the
+        #      order which should be searched in the local system cache location.
+        #      When an item isn't found it will search for archive files which prefix matches
+        #      the item in the list. If more than one archive file is matched for a prefix
+        #      the archive file most recently modified will be used. If there is no
+        #      matching archive file then nothing will be restored in the docker container.
+        #
+        #    Save Cache:
+        #      The first local cache key in the list is used for the name of the local
+        #      cache archive file
+        #
         caches:
+          <local cache key>: <docker path>
+
           maven: "/root/.m2/repository"
 
+          <docker path>:
+            - <local cache key A>
+            - <local cache key B>
+
+          "/root/.m2/venv"
+            - venv-{{ checksum(["requirements.txt", "test_requirements.txt"])
+            - venv-
```

Comment:
venv and .m2 are mutually exclusive :) venv is for python and .m2 is for maven.

Maybe you can change it to this:

```
"/root/.m2/repository":
# Buildrunner will look for a cache that matches this cache key/prefix, typically the first key should be the most specific as it is the closest match
# Note that this first key will also be used to save the cache for use across builds or projects
- m2repo-{{ checksum(["pom.xml", "subproj/pom.xml"]) }}
# If the first cache key is not found in the caches, use this prefix to look for a cache that may not be an exact match,
# but may still be close and not require as much downloading of dependencies, etc
# Note that this may match across any cache done by any build on the same system, so it may be wise to
# use a unique prefix for any number of builds that have a similar dependency tree, etc
- m2repo-
# If no cache is found, nothing will be extracted and the application will need to rebuild the cache from scratch
```

Reply:
- [ ] Update comment to be more representative of how it would be used.